allennlp/semparse/contexts/table_question_context.py

```diff
@@ -0,0 +1,307 @@
+        'two': 2,
+        'three': 3,
+        'four': 4,
+        'five': 5,
+        'six': 6,
+        'seven': 7,
+        'eight': 8,
+        'nine': 9,
+        'ten': 10,
+        'first': 1,
+        'second': 2,
+        'third': 3,
+        'fourth': 4,
+        'fifth': 5,
+        'sixth': 6,
+        'seventh': 7,
+        'eighth': 8,
+        'ninth': 9,
+        'tenth': 10,
+        **MONTH_NUMBERS,
+        }
+
+
+class TableQuestionContext:
+    """
+    A Barebones implementation similar to https://github.com/crazydonkey200/neural-symbolic-machines/blob/master/table/wtq/preprocess.py
+    for extracting entities from a question given a table and type its columns with <string> | <date> | <number>
+    """
+
+    def __init__(self,
+                 cell_values: Set[str],
+                 column_type_statistics: List[Dict[str, int]],
+                 column_index_to_name:  Dict[int, str]) -> None:
+        self.cell_values = cell_values
+        self.column_types = {column_index_to_name[column_index] : max(column_type_statistics[column_index], 
+                                           key=column_type_statistics[column_index].get) for column_index in column_index_to_name}
+
+    @classmethod
+    def read_from_lines(cls, lines: List[str], max_tokens_for_num_cell : int) -> 'TableQuestionContext':
+        column_index_to_name = {}
+
+        header = lines[0] # the first line is the header
+        index = 1
+        while lines[index][0] == '-1':
+            # column names start with fb:row.row.
+            curr_line = lines[index]
+            column_name_sempre = curr_line[2]
+            column_index = int(curr_line[1])
+            column_name = column_name_sempre.replace('fb:row.row.', '')
+            column_index_to_name[column_index] = column_name
+            index += 1
+
+
+        column_node_type_info = [{'string' : 0, 'number' : 0, 'date' : 0}
+                                 for col in column_index_to_name]
+        cell_values = set()
+        while index < len(lines):
+            curr_line = lines[index]
+            column_index = int(curr_line[1])
+            node_info = dict(zip(header, curr_line))
+            cell_values.add(cls._normalize_string(node_info['content']))
+            num_tokens = len(node_info['tokens'].split('|'))
+            if node_info['date']:
+                column_node_type_info[column_index]['date'] += 1
+            # If cell contains too many tokens, then likely not number
+            elif node_info['number'] and num_tokens < max_tokens_for_num_cell:
+                column_node_type_info[column_index]['number'] += 1
+            elif node_info['content'] != '—':
+                column_node_type_info[column_index]['string'] += 1
+            index += 1
+
+
+        return cls(cell_values, column_node_type_info, column_index_to_name)
+
+    @classmethod
+    def read_from_file(cls, filename: str, max_tokens_for_num_cell: int) -> 'TableQuestionContext':
```

Comment:
Seems like `max_tokens_for_num_cell` needs to be a global decision instead of having the user pass this value every time this method is called.

Reply:
When should this value ever be greater than 1? Is there are problem with always considering values with more than one token as non-numbers?